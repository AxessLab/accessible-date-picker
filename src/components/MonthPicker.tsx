import React from "react";
import moment from "moment";
import styles from "../styles/stylesCalendar.css";

interface SetDateObject {
    year: number;
    month: number;
    dates: number;
}
interface MonthPickerProps {
    month: number;
    year: number;
    setDateObject: (object: SetDateObject) => void;
}

const monthArray = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];

const MonthPicker: React.FC<MonthPickerProps> = (props) => {
    const { month, year, setDateObject } = props

    type changeMonthParameter = "next" | "previous";
    const changeMonthHandler = (selection: changeMonthParameter) => {
        let selectedYear = year;
        let selectedMonth = month;

        if (selection === "next") {
            if (selectedMonth + 1 === 12) {
                selectedYear++;
                selectedMonth = 0;
            } else {
                selectedMonth++;
            }
        }

        if (selection === "previous") {
            if (selectedMonth - 1 >= 0) {
                selectedMonth--;
            } else {
                selectedYear--;
                selectedMonth = 11;
            }
        }

        const selectedDates = +moment(`${selectedYear}-${selectedMonth + 1}`, "YYYY-MM").daysInMonth();

        setDateObject({
            year: selectedYear,
            month: selectedMonth,
            dates: selectedDates
        });

    };

    return (
        <div>
            <button className={`${styles.previousButton} ${styles.monthSelectionButtons}`} aria-label="previous month" onClick={() => { changeMonthHandler("previous") }}>Previous</button>
            <h2 id="month-label" aria-live="assertive">{monthArray[month]}{" "}{year}</h2>
            <button className={`${styles.nextButton} ${styles.monthSelectionButtons}`} aria-label="next month" onClick={() => { changeMonthHandler("next") }}>Next</button>
        </div>
    );
};

export default MonthPicker;