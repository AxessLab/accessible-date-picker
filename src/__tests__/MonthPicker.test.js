import React from "react";
import { shallow, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';
import MonthPicker from "../components/MonthPicker";
import { ThemeProvider } from "react-jss";

configure({ adapter: new Adapter() });

const currentDate = {
    year: 2021,
    month: 0,
    dates: 31
};
const clickedDate = {}
const mockSetDateObject = jest.fn();
const theme = {
    palette: {
        primary: "#f5f5f5",
        secondary: "#2b4450",
        tertiary: "#871111"
    },
    spacing: ["0px", "4px", "8px", "16px", "32px", "64px"],
};

const wrapper = shallow(<ThemeProvider theme={theme}><MonthPicker currentDate={currentDate} clickedDate={clickedDate} setDateObject={mockSetDateObject} /></ThemeProvider>);

test("renders correct heading that changes with next/previous", () => {
    wrapper.dive().find('#button-next').simulate('click');
    expect(mockSetDateObject).toHaveBeenCalled();

    wrapper.find('#button-previous').simulate('click');
    expect(mockSetDateObject).toHaveBeenCalled();

    const h2 = wrapper.find('#month-label').text();
    expect(h2).toEqual("January 2021");

    expect(toJson(wrapper)).toMatchSnapshot();
});

