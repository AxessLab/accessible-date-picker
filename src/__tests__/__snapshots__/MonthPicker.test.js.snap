// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct heading that changes with next/previous 1`] = `
<div
  aria-live="assertive"
>
  <button
    aria-label="previous month"
    className="previousCalendarButton monthSelectionButtons"
    id="button-previous"
    onClick={[Function]}
  >
    Previous
  </button>
  <h2
    className="h2DatePicker"
    id="month-label"
  >
    January
     
    2021
  </h2>
  <button
    aria-label="next month"
    className="nextCalendarButton monthSelectionButtons"
    id="button-next"
    onClick={[Function]}
  >
    Next
  </button>
</div>
`;
